### PR TITLE
added configurable booking prefix

### PR DESCRIPTION
**Pull request by DFSko**
added configurable booking prefix

resolves happy-travel/agent-app-project#77

### CODE REVIEW

File: Api/Infrastructure/Options/BookingOptions.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace HappyTravel.Edo.Api.Infrastructure.Options
+{
+    public class BookingOptions
```
Comment (mingazhev) [inline Api/Infrastructure/Options/BookingOptions.cs:3]: This is not booking options, it is more like tag or reference code options. Reference codes may apply not only for the booking

File: Api/Services/CodeProcessors/TagProcessor.cs [hunk 2]
```diff
@@ -2,27 +2,35 @@
...
+            {
+                _bookingOptions.ReferenceCodePrefix,
+                serviceType.ToString(),
                 destinationCode,
                 itineraryNumber,
-                currentNumber.ToString("D2"));
+                currentNumber.ToString("D2")
+            };
+
+            return string.Join(ReferenceCodeItemsSeparator, values.Where(v => v is not null));
```
Comment (mingazhev) [inline Api/Services/CodeProcessors/TagProcessor.cs:33]: Would it be null there? Maybe better to use somthing like string.IsNullOrWhitespace?

File: Api/appsettings.json [hunk 3]
```diff
@@ -71,6 +71,9 @@
     "Payfort": {
       "Options": "edo/payfort",
       "Urls": "edo/payfort/urls"
+    },
```
Comment (mingazhev) [inline Api/appsettings.json:74]: Same there, tag processing is not bound to bookings directly

File: HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs [hunk 4]
```diff
@@ -43,6 +48,47 @@ EdoContext CreateDbContextWithDuplicateItnValidation()
                 return context.Object;
             }
         }
+
+
+        [InlineData("DEV", ServiceTypes.HTL, "ABC", "DEV-HTL-ABC-000001-00")]
+        [InlineData(null, ServiceTypes.HTL, "BCA", "HTL-BCA-000001-00")]
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs:54]: I'm pressing `F` to pay respect for the tests

File: HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs [hunk 5]
```diff
@@ -43,6 +48,47 @@ EdoContext CreateDbContextWithDuplicateItnValidation()
...
+                .Returns(() => Task.CompletedTask);
+            
+            var tagProcessor = new TagProcessor(context.Object, Options.Create(new BookingOptions
+            {
+                ReferenceCodePrefix = prefix
+            }));
+            
+            var itn = await tagProcessor.GenerateItn();
+
+            var referenceCode = await tagProcessor.GenerateReferenceCode(serviceType, destinationCode, itn);
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs:72]: Could you please additionally check this for non-sequential RefCode?

File: HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs (hunk 4 above)
Comment (kirillta) [reply HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs:54]: `F`

File: Api/Infrastructure/Options/BookingOptions.cs (hunk 1 above)
Comment (DFSko) [reply Api/Infrastructure/Options/BookingOptions.cs]: Renamed

File: Api/Services/CodeProcessors/TagProcessor.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/CodeProcessors/TagProcessor.cs]: Fixed

File: HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs (hunk 5 above)
Comment (DFSko) [reply HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs]: Added

File: HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs [hunk 6]
```diff
@@ -43,6 +48,40 @@ EdoContext CreateDbContextWithDuplicateItnValidation()
...
+            {
+                ReferenceCodePrefix = prefix
+            }));
+            
+            var itn = await tagProcessor.GenerateItn();
+
+            var sequentialReferenceCode = await tagProcessor.GenerateReferenceCode(ServiceTypes.HTL, "ABC", itn);
+            var nonSequentialReferenceCode = await tagProcessor.GenerateNonSequentialReferenceCode(ServiceTypes.HTL, "ABC");
+            
+            Assert.Equal($"{result}-00", sequentialReferenceCode);
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs:75]: It's better to write three different test methods there. Three different methods under one test is not a good option

Comment (DFSko) [reply HappyTravel.Edo.UnitTests/Tests/Services/CodeProcessors/TagProcessorTests/ItnNumbersGeneration.cs]: Fixed

**Review by kirillta — APPROVED** [review]

**Review by mingazhev — APPROVED** [review]

**Pull request merged**